### PR TITLE
Changes to EOD reports

### PR DESCRIPTION
**Pull request by teryhill**
Added user names to the eod reports, set them to look at the encounter
provider (helps eliminate provider 0 issues) added payment type. Added a
4th report that keys off a user field in the fee sheet and gives a total
for the user that did the fee sheet. #1143

### CODE REVIEW

**Comment by teryhill** [comment]
@tmccormi and @aethelwulffe  ready for y'all to look at this

File: interface/billing/print_daysheet_report_num1.php [hunk 1]
```diff
@@ -162,20 +162,27 @@
 // this loop gathers the user and provider numbers
 foreach ($ret as $iter) {
   $catch_user[] = $iter{'user'};
-  $catch_provider[] = $iter{'provider_id'};
+  $catch_provider[] = $iter{'enc_provider_id'};
+  if($iter{'code_type'} == 'Patient Payment' && ($iter{'paytype'} == 'PCP' || $iter{'paytype'} == 'PP') )
```
Comment (aethelwulffe) [inline interface/billing/print_daysheet_report_num1.php]: remember "code_type" has providence elsewhere (codes table)

File: interface/billing/print_daysheet_report_num1.php [hunk 2]
```diff
@@ -162,20 +162,27 @@
...
+    $catch_payment_method[]  = $iter{'pay_method'};
```
Comment (aethelwulffe) [inline interface/billing/print_daysheet_report_num1.php]: Copays?

File: interface/billing/print_daysheet_report_num1.php [hunk 3]
```diff
@@ -190,735 +197,822 @@
...
         }
         if (($iter{'ins_code'}) < 0 AND ($iter{'code_type'}) === 'Insurance Payment' ){
-          $us0_insref = $us0_insref + $iter{'ins_code'};        
+          $us0_insref = $us0_insref + $iter{'ins_code'};
         }
-        if (($iter{'pat_code'}) < 0 AND ($iter{'code_type'}) === 'Patient Payment' AND $iter{'paytype'} != 'PCP' ){ 
+        if (($iter{'pat_code'}) < 0 AND ($iter{'code_type'}) === 'Patient Payment' AND $iter{'paytype'} != 'PCP' ){
           $us0_patref = $us0_patref + $iter{'pat_code'};
         }
+
```
Comment (aethelwulffe) [inline interface/billing/print_daysheet_report_num1.php]: if ($iter{'paytype'})=='PP'{} here maybe

Another would be use  foreach($case as $arg => $v){

File: interface/billing/print_daysheet_report_num1.php [hunk 4]
```diff
@@ -1581,60 +1717,62 @@
...
+    print  "<td width=25><span class=text>";
+     $prov_name = sqlQuery("SELECT lname, fname FROM users WHERE ".
+     "id = ?",array($provider_info['user'][$i]));
+    Printf ("</span></td><td width=250><span class=text><center>".text($provider_info[user][$i]) . ' ' . $prov_name['fname'] . ' ' . $prov_name['lname']) . "</center>";
+    print  "</span></td><td width=125><span class=text>";
     printf ("</span></td><td width=250><span class=text><b>" . xlt("Total Charges").': '." %1\$.2f ", text($provider_info[fee][$i])). "</b>";
-    print  "</span></td><td width=125><span class=text>";   
-    printf ("</span></td><td width=250><span class=text><b>"  . xlt("Total Payments").': '. "(%1\$.2f)",text($provider_info[inspay][$i] + $provider_info[patpay][$i])) . "</b>";    
-    print  "</span></td><td width=25><span class=text>";        
+    print  "</span></td><td width=125><span class=text>";
```
Comment (aethelwulffe) [inline interface/billing/print_daysheet_report_num1.php]: The username addition looks fine.

File: interface/billing/print_daysheet_report_num2.php [hunk 5]
```diff
@@ -435,29 +435,29 @@
...
-     if ($iter{'paytype'} === 'PCP') { 
-      print  "</span></td><td width=250><span class=text><center>" . xlt('COPAY'). "</center>";
+     if ($iter{'paytype'} === 'PCP') {
+      print  "</span></td><td width=250><span class=text><center>" . xlt('COPAY')." / " . text($iter{'pay_method'}). "</center>";
      }
      if (($iter{'code_type'}) != 'Insurance Payment' AND ($iter{'code_type'}) != 'Patient Payment' AND $iter{'paytype'} != 'PCP') {
          print  "</span></td><td width=100><span class=text><center>" . text($iter{'code_type'}). "</center>";
      }
-      print  "</span></td><td width=100><span class=text><center>" . text($iter{'provider_id'}). "</center>"; 
+      print  "</span></td><td width=100><span class=text><center>" . text($iter{'enc_provider_id'}). "</center>";
```
Comment (aethelwulffe) [inline interface/billing/print_daysheet_report_num2.php]: Be good to have somewhere to configure the report for what field to check.  Thinking of referring, supervising, and interns.

File: interface/forms/fee_sheet/new.php [hunk 6]
```diff
@@ -1341,8 +1341,12 @@ function newEvt() {
 
 if ($GLOBALS['contract_physician_in_feesheet']) {
   echo "&nbsp;&nbsp;" . xlt($GLOBALS['contract_physician_in_feesheet_name']) . "\n";
+  if (trim($GLOBALS['contract_physician_in_feesheet_name']) == 'Users') {
```
Comment (aethelwulffe) [inline interface/forms/fee_sheet/new.php]: I'm going to have to figure out how to convert my system to use this efficiently.  ARSP has that "counselor_id" thing where it checks pay rates and all that.  I will try to help blend the functions in the near future.

**Comment by aethelwulffe** [comment]
I honestly can't fully test this at the moment.  I will get a prod instance copy to do a look-see.  Any EOD report options I should select?

File: interface/billing/print_daysheet_report_num1.php (hunk 2 above)
Comment (teryhill) [reply interface/billing/print_daysheet_report_num1.php]: image
 they are a payment method

Comment (aethelwulffe) [reply interface/billing/print_daysheet_report_num1.php]: Ah, not a "paytype".  Gotcha.

**Comment by tmccormi** [comment]
One thing I would like to see (someday) is for these reports to just be "reports" not buried in EOB/Billing and for ALL 4 to be runnable without changing globals.

**Comment by teryhill** [comment]
Approved x2 Merging

**Pull request merged**